documentation/modules/ROOT/partials/modules/all-connectors/con-connector-ad-hoc-snapshots.adoc

```diff
@@ -0,0 +1,93 @@
+For more information about support scope, see link:https://access.redhat.com/support/offerings/techpreview/[Technology Preview Features Support Scope].
+====
+endif::product[]
+
+By default, a connector runs an initial snapshot operation only after it starts for the first time.
+Following this initial snapshot, under normal circumstances, the connector does not repeat the snapshot process.
+Any future change event data that the connector captures comes in through the streaming process only.
+
+However, in some situations the data that the connector obtained during the initial snapshot might become stale, lost, or incomplete.
+To provide a mechanism for recapturing {data-collection} data, {prodname} includes an option to perform ad hoc snapshots.
+The following changes in a database might be cause for performing an ad hoc snapshot:
+
+* The connector configuration is modified to capture a different set of {data-collection}s.
+* Kafka topics are deleted and must be rebuilt.
+* Data corruption occurs due to a configuration error or some other problem.
+
+You can re-run a snapshot for a {data-collection} for which you previously captured a snapshot by initiating a so-called _ad-hoc snapshot_.
+Ad hoc snapshots require the use of {link-prefix}:{link-signalling}#sending-signals-to-a-debezium-connector[signaling {data-collection}s].
+You initiate an ad hoc snapshot by sending a signal request to the {prodname} signaling {data-collection}.
+
+When you initiate an ad hoc snapshot of an existing {data-collection}, the connector appends content to the topic that already exists for the {data-collection}.
+If a previously existing topic was removed, {prodname} can create a topic automatically if xref:{link-topic-auto-creation}#customizing-debezium-automatically-created-topics[automatic topic creation] is enabled.
+
+Ad hoc snapshot signals specify the {data-collection}s to include in the snapshot.
+The snapshot can capture the entire contents of the database, or capture only a subset of the {data-collection}s in the database.
+
+You specify the {data-collection}s to capture by sending an `execute-snapshot` message to the signaling {data-collection}.
+Set the type of the `execute-snapshot` signal to `incremental`, and provide the names of the {data-collection}s to include in the snapshot, as described in the following table:
+
+.Example of an ad hoc `execute-snapshot` signal record
+[cols="2,2,6",options="header"]
+|===
+|Field | Default | Value
+
+|`type`
+|`incremental`
+| Specifies the type of snapshot that you want to run. +
+Setting the type is optional.
+Currently, you can request only `incremental` snapshots.
+
+
+|`data-collections`
+|_N/A_
+| An array that contains the fully-qualified names of the {data-collection} to be snapshotted. +
+The format of the names is the same as for the `signal.data.collection` configuration option.
+
+|===
+
+.Triggering an ad hoc snapshot
```

Comment:
This should be probably named `Triggering an ad hoc incremental snapshot` as the text descibes incrmental snapshot specifics.

Reply:
I was torn about this. Given that we don't currently provide for another type of ad hoc snapshot, I wasn't sure what to do. Maybe I"ll remove the section and just link to the incremental snapshot topic?

Reply:
I've commented out the procedure and example in this section, leaving only the information about how the current  `execute-snapshot` action is used for incremental snapshots only. I've provided a cross-reference to the contextual incremental snapshot topic. I might have to remove that if it doesn't work in the build. 🤞🏼